main.py

```diff
@@ -356,7 +359,10 @@ def run(self) -> None:
             for future in concurrent.futures.as_completed(futures):
                 test_run = future.result()
                 self.count_result(test_run)
-                if self.verbose:
+                if self.verbose or (self.fail_only and test_run.result != TestResult.PASSED):
```

Comment:
Perhaps we should also exclude skipped tests here? I wouldn't consider them "failing".

Reply:
Good point I added `NON_FAIL_RESULTS = [TestResult.PASSED, TestResult.SKIPPED]` at the top to be able to specify the non failing results (for this mode only).